patches/graphql-relay+0.5.3.patch

```diff
@@ -0,0 +1,31 @@
+patch-package
+--- a/node_modules/graphql-relay/lib/connection/arrayconnection.js
++++ b/node_modules/graphql-relay/lib/connection/arrayconnection.js
+@@ -61,7 +61,8 @@ function connectionFromArraySlice(arraySlice, args, meta) {
+       first = args.first,
+       last = args.last;
+   var sliceStart = meta.sliceStart,
+-      arrayLength = meta.arrayLength;
++      arrayLength = meta.arrayLength,
++      resolveNode= meta.resolveNode;
+ 
+   var sliceEnd = sliceStart + arraySlice.length;
+   var beforeOffset = getOffsetWithDefault(before, arrayLength);
+@@ -88,10 +89,15 @@ function connectionFromArraySlice(arraySlice, args, meta) {
+   var slice = arraySlice.slice(Math.max(startOffset - sliceStart, 0), arraySlice.length - (sliceEnd - endOffset));
+ 
+   var edges = slice.map(function (value, index) {
+-    return {
++    const newEdge = {
+       cursor: offsetToCursor(startOffset + index),
+-      node: value
++      node: !!resolveNode ? Promise.resolve(resolveNode(value)) : value,
+     };
++
++    if (resolveNode) {
++      return Object.assign({ ...value }, newEdge, {});
++    }
++    return newEdge;
```

Comment:
Not sure if you missed this earlier, so repeating here to be sure:

```js
return resolveNode ? { ...value, ...newEdge } : newEdge;
```

Reply:
I did see! I was actually wondering, I noticed this `*.js` file doesn't use some of the ES2015 that Metaphysics has such destructuring, spread, etc. and was wondering if I should.

I figured the spread here worked since our babel is setup, but wasn't sure the best practice. I'll play around with your suggestion.

Reply:
Good good 👍